### DISCUSSION TITLE
How can I set the collection order?

### DISCUSSION THREAD

**Discussion by devaniljr** [Shopify/maintenance_tasks#928] [Q&A]
Maybe I'm missing something obvious, but I'm trying to create my collection in descending order using `Model.where(conditions).order(created_at: :desc)`, but it gives me this error:

> JobIteration::ActiveRecordCursor::ConditionNotSupportedError
> 
> The relation cannot use ORDER BY or LIMIT due to the way how iteration with a cursor is designed. You can use other ways to limit the number of rows, e.g. a WHERE condition on the primary key column.

How can I set the order of my collection?

**Comment by etiennebarrie** [comment]
You cannot because we need to be able to iterate over records efficiently, and the way to do that is to go through records in primary key order, keeping the id as a cursor and using WHERE and LIMIT instead of OFFSET.
